### PR TITLE
coccinelle: add a test

### PR DESCRIPTION
**Pull request by Moisan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Add a test to coccinelle. Related to #11898.

### CODE REVIEW

File: Formula/coccinelle.rb [hunk 1]
```diff
@@ -32,5 +32,18 @@ def install
                           "--prefix=#{prefix}"
     system "opam", "config", "exec", "--", "make"
     system "make", "install"
+
+    share.install ["demos/simple.cocci", "demos/simple.c"]
```
Comment (ilovezfs) [inline Formula/coccinelle.rb]: should use pkgshare

File: Formula/coccinelle.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@ class Coccinelle < Formula
   homepage "http://coccinelle.lip6.fr/"
   url "http://coccinelle.lip6.fr/distrib/coccinelle-1.0.6.tgz"
   sha256 "8452ed265c209dae99cbb33b67bc7912e72f8bca1e24f33f1a88ba3d7985e909"
-  revision 1
+  revision 2
```
Comment (ilovezfs) [inline Formula/coccinelle.rb]: this is a small enough change not to warrant a revision bump because the existing installations aren't actually broken

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/coccinelle.rb [hunk 3]
```diff
@@ -33,11 +33,11 @@ def install
     system "opam", "config", "exec", "--", "make"
     system "make", "install"
 
-    share.install ["demos/simple.cocci", "demos/simple.c"]
+    pkgshare.install ["demos/simple.cocci", "demos/simple.c"]
   end
 
   test do
-    system "#{bin}/spatch", "-sp_file", "#{share}/simple.cocci", "#{share}/simple.c", "-o", "new_simple.c"
+    system "#{bin}/spatch", "-sp_file", "#{pkgshare}/simple.cocci", "#{pkgshare}/simple.c", "-o", "new_simple.c"
```
Comment (ilovezfs) [inline Formula/coccinelle.rb]: Please wrap split this into two lines so it doesn't go passed the eightieth column like this:
```
    system "#{bin}/spatch", "-sp_file", "#{pkgshare}/simple.cocci",
                            "#{pkgshare}/simple.c", "-o", "new_simple.c"
```

File: Formula/coccinelle.rb [hunk 4]
```diff
@@ -32,5 +32,18 @@ def install
                           "--prefix=#{prefix}"
     system "opam", "config", "exec", "--", "make"
     system "make", "install"
+
+    pkgshare.install ["demos/simple.cocci", "demos/simple.c"]
```
Comment (ilovezfs) [inline Formula/coccinelle.rb]: you don't need the array

File: Formula/coccinelle.rb [hunk 5]
```diff
@@ -32,5 +32,19 @@ def install
...
     system "opam", "config", "exec", "--", "make"
     system "make", "install"
+
+    pkgshare.install "demos/simple.cocci", "demos/simple.c"
+  end
+
+  test do
+    system "#{bin}/spatch", "-sp_file", "#{pkgshare}/simple.cocci",
+                            "#{pkgshare}/simple.c", "-o", "new_simple.c"
+    correct_output = <<~EOS
```
Comment (ilovezfs) [inline Formula/coccinelle.rb]: use `expected` instead of `correct_output`

File: Formula/coccinelle.rb [hunk 6]
```diff
@@ -32,5 +32,19 @@ def install
...
+      int main(int i) {
+        f("ca va", 3);
+        f(g("ca va pas"), 3);
+      }
+    EOS
+    assert_equal correct_output, File.read(testpath/"new_simple.c")
```
Comment (ilovezfs) [inline Formula/coccinelle.rb]: Use `(testpath/"new_simple.c").read`

**Comment by ilovezfs** [comment]
@Moisan a couple more nits noted above, and then could you please squash this down to one commit? Thanks!

**Review by ilovezfs — APPROVED** [review]

**Comment by woodruffw** [comment]
Thanks @Moisan!

**Pull request closed without merging**